lib/kino/download.ex

```diff
@@ -0,0 +1,46 @@
+defmodule Kino.Download do
+  @moduledoc ~S'''
+  Built on top of `Kino.JS`, uses Javascript to initiate a download
+  on the client's machine.
+  '''
+  use Kino.JS
+
+  @doc ~S'''
+  Initiates a download on the client's machine.  The content is
+  generated lazily by invoking the `content_fun` argument.
+
+  ## Examples
+
+  Passing the content from an existing variable:
+
+      json = ~s/{"foo": "bar"}/
+      Kino.Download.start("file.json", fn -> json end)
+
+  Lazily generating a large file:
+
+      Kino.Download.start("large-file.txt", fn ->
+        String.duplicate("data ", 10_000_000)
+      end)
+  '''
+  @spec start(String.t(), (() -> String.t())) :: Kino.JS.t()
+  def start(filename, content_fun) do
+    data = %{
+      content: content_fun.(),
+      filename: filename
+    }
+
+    Kino.JS.new(__MODULE__, data)
+  end
+
+  asset "main.js" do
+    """
+    export function init(ctx, data) {
+      var hiddenElement = document.createElement('a');
+      hiddenElement.href = 'data:attachment/text,' + encodeURI(data.content);
```

Comment:
Annotation: I need to make this `attachment/text` dynamic.  My problem is that I did not know how to test it.  I tried testing this with `csv`, `json`, and plain text files and leaving the type as `attachment/text` never seemed to break anything.

Reply:
To make it more generic, we can receive any binary from the server, then encode it as base64 and use `data:application/octet-stream;base64;${dataBase64}`. If we need to customize the content type explicitly, we can make it an option in Elixir, but I think we should be fine with `application/octet-stream`, especially that we specify the file name.